server/apps/users/forms.py

```diff
@@ -12,47 +24,57 @@ class UserProfileForm(forms.Form):
         ("unspecified", "Prefer not to say"),
     ]
     age_bracket = forms.ChoiceField(choices = age_brackets, widget = forms.Select(), required=False,
-                                   label="What is your age?",
+                                   label="What is your age group?",
+                                   help_text="Providing your age will help researchers understand potential trends across age groups. It will also help readers to better understand your perspective if this data is made public.",
                                    initial="unspecified")
     age_bracket.group = 1
 
+    age_public = forms.BooleanField(label = "Make age public",
+        required=False,
+        widget=forms.CheckboxInput(attrs={"class": "custom-control-input"}))
+    age_public.group = 1
+    age_public.gap = True
+
     genders = [
         ("nonbinary", "Non-binary"),
         ("female", "Female"),
         ("male", "Male"),
-        ("see_description", "See description"),
+        ("self_identify", "Self identify"),
         ("unspecified", "Prefer not to say"),
     ]
     gender = forms.ChoiceField(choices = genders, widget = forms.Select(), required=False,
-                              label="Which gender do you identify as?",
+                              label="What gender do you identify with?",
+                              help_text="Providing your gender identity will help researchers understand potential trends across this demographic. It will also help readers to better understand your perspective if this data is made public.",
                               initial="unspecified")
     gender.group = 1
 
-    # Not intended as final
-    autistic_identifications = [
-        ("yes", "Yes"),
-        ("no", "No"),
-        ("not_sure", "Not sure"),
-        ("self_describe", "Prefer to self-describe"),
-        ("unspecified", "Prefer not to say"),
-    ]
-    autistic_identification = forms.ChoiceField(choices = autistic_identifications, widget = forms.Select(),
-                                           label="Do you identify as autistic?",
-                                           required=False, initial="unspecified")
-    autistic_identification.group = 1
 
-    description = forms.CharField(label="Description", max_length=2048, strip=True, required=False,
-                                 widget=forms.Textarea(attrs={"placeholder":"An optional description of yourself",
+    gender_self_identification = forms.CharField(label="Self identification",
+                                 max_length=150, strip=True, required=False,
+                                 widget=forms.TextInput())
+
+    gender_self_identification.group = 1
+    gender_self_identification.layout_horizontal = True
```

Comment:
The JS hide/show of the free-text field for self-identification works. But what doesn't seem to work (at least for me in Firefox) is actually filling in the text. The JS console doesn't give any error or anything, but I can't seem to actually type into the box 🙈

Reply:
Ooops, thanks for checking. An error in refactoring meant the disabled flag got set in reverse. It should be fixed now.